controllers/useraccount/useraccount_controller.go

```diff
@@ -385,13 +385,29 @@ func setOwnerAndProviderLabel(object metav1.Object, owner string) {
 	object.SetLabels(labels)
 }
 
-func addProviderLabel(object client.Object, cl client.Client) error {
+func addOwnerAndProviderLabels(object client.Object, cl client.Client, owner string) error {
+	var changed bool
 	if _, exists := object.GetLabels()[toolchainv1alpha1.ProviderLabelKey]; !exists {
 		labels := object.GetLabels()
 		if labels == nil {
 			labels = make(map[string]string)
 		}
 		labels[toolchainv1alpha1.ProviderLabelKey] = toolchainv1alpha1.ProviderLabelValue
+		object.SetLabels(labels)
+		changed = true
+	}
+
+	if _, exists := object.GetLabels()[toolchainv1alpha1.OwnerLabelKey]; !exists {
+		labels := object.GetLabels()
+		if labels == nil {
+			labels = make(map[string]string)
+		}
```

Comment:
it's just a detail, but this case shouldn't ever happen - if the labels are nil, then they would be set in the first case (when checking provider label)